### ISSUE TITLE
Use  . (dot) as an option in makeFun()

### ISSUE THREAD

**Issue by dtkaplan** [ProjectMOSAIC/mosaicCalc#9]
Things are still a bit of a mess with the order of arguments created by `makeFun()`, `D()`, and `antiD()`. For instance, `makeFun()` will use the symbols to the right of tilde as the arguments to the function created, in the order specified. But parameters might come in any order, depending on how the tilde expression is composed.  `D()` will insert its "with respect to" argument in the first argument position, which means that `D(x*y^2 ~x)` and `D(x*y^2~y)` will have the arguments in different orders. 

I'm not entirely sure how to handle this in `D()`. Perhaps encourage a style where one writes `f <- makeFun(x*y^2 ~ x + y)` and then `D(f(x,y) ~ y)`. Right now, this is awkward, because `D()` doesn't know to look inside a non-standard function such as `f()` for a symbolic formula. Instead, it applies numerical differentiation. Good to fix that.

But I'm thinking it might be nice to have an option in `makeFun()` that lets you write `makeFun(x*y^2 ~ .)` or similar, so that `makeFun()` knows to choose a canonical ordering of the arguments.

**Comment by nicholasjhorton** [comment]
@dtkaplan would it be useful to find a time to talk about this and any other related mosaic package suite issues?

**Comment by dtkaplan** [comment]
Yes, absolutely. I'd love to talk about this and some of the things going
on in the mosaicCalc world.  Any time.

- Danny
651-295-7487

**Comment by rpruim** [comment]
Can we find a time next week?

**Comment by nicholasjhorton** [comment]
I'm teaching a workshop next week as part of the DSC-WAV project, so I can't meet next week. 

That being said, I wonder if the two of you might start a discussion since I've not been as directly involved.

**Comment by dtkaplan** [comment]
Perhaps just add a `canonical_order(f)` which will re-arrange arguments according to (x, y, z, u, v, w, n, p, then alphabetical). Just use this in `D()`.
